client/src/components/question/ThirdPartyIcons.js

```diff
@@ -95,36 +100,44 @@ class ThirdPartyIcons extends React.Component {
     const googleImagesPrefix = 'https://google.com/search?tbm=isch&q=';
     const typePlural = question.type === "tossup" ? "tossups" : "bonuses";
 
+	//wrapper function that produces an icon with an associated ReactTooltip
+	//goes without saying but don't put any spaces in your name variable
+	const withTooltip = (name, onClick, tooltip) => <>
+      <Icon name={name} className='icon-clickable'
+            onClick={onClick}/>
+      <ReactTooltip effect='solid' type='info' id={`${question.id}-${name}`}>
+	    {tooltip}
+      </ReactTooltip>
+	</>;
+	
+	//too lazy to code special case for admin button - just split list into before and after admin button
+	const iconsBeforeAdmin = [
+	  {name: 'google', onClick: () => handleIconClick(googlePrefix, query), tooltip: 'Google this answerline'},
+	  {name: 'image', onClick: () => handleIconClick(googleImagesPrefix, query), tooltip: 'Google Image search this answerline'},
+	  {name: 'wikipedia', onClick: () => handleWikiIconClick(question, index), tooltip: 'Search for this answerline on Wikipedia'}
+	];
+	
+	const iconsAfterAdmin = [
+	  {name: 'clone', onClick: () => handleIconClick("copy", query), tooltip: 'Copy this answerline'},
+	  {name: 'repeat', onClick: () => handleSearchIconClick(query), tooltip: 'Search for this answerline, with no filters'},
+	  {name: 'refresh', onClick: () => handleSearchIconClick(query, false), tooltip: 'Search for this answerline, with the same filters'}
+	];
+
     return (
       <Grid.Column largeScreen='3' computer='2' tablet='16' mobile='16'
         verticalAlign='middle'
         textAlign='center'
         className='question-icons'
       >
-        <Icon name='google' className='icon-clickable'
-              onClick={() => handleIconClick(googlePrefix, query)}/>
-        <Icon corner name='image' className='icon-clickable'
-              onClick={() => handleIconClick(googleImagesPrefix, query)}/>
-        <Icon name='wikipedia w' className='icon-clickable'
-              onClick={() => handleWikiIconClick(question, index)}/>
+		{iconsBeforeAdmin.map((icon) => withTooltip(...icon))} //I don't know if spread works here - if not, just get all the params manually
```

Comment:
It shouldn't, your target spread is ordered parameters, but you're spreading an object -- for this to work you would either change `iconsXAdmin` into an array of arrays, or you'd change `withTooltip` to take named parameters (and not bother with the spread)